main.go

```diff
@@ -1,133 +1,151 @@
-package main
-
-import (
-	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"log"
-	"net/http"
-	"sync"
-	"strconv"
-
-	"github.com/gorilla/mux"
-)
-
-type autoInc struct {
-    sync.Mutex
-    id int
-}
-
-func (a *autoInc) ID() (id int) {
-    a.Lock()
-    defer a.Unlock()
-
-    id = a.id
-    a.id++
-    return
-}
-
-var unique autoInc
-
-type event struct {
-	ID          int		`json:"ID"`
-	Title       string	`json:"Title"`
-	Description string	`json:"Description"`
-}
-
-var events = []event{
-	{
-		ID:          unique.ID(),
-		Title:       "Introduction to Golang",
-		Description: "Come join us for a chance to learn how golang works and get to eventually try it out",
-	},
-}
-
-func createEvent(w http.ResponseWriter, r *http.Request) {
-	var newEvent event
-	newEvent.ID = unique.ID()
-
-	reqBody, err := ioutil.ReadAll(r.Body)
-    if err != nil {
-        http.Error(w, err.Error(), http.StatusBadRequest)
-        return
-    }
-	json.Unmarshal(reqBody, &newEvent)
-	events = append(events, newEvent)
-	w.WriteHeader(http.StatusCreated)
-
-	json.NewEncoder(w).Encode(newEvent)

```

Comment:
It's unclear to me if we need the `w.WriteHeader()` stuff or not. It seems good to do? So I added it elsewhere